mp2-v1/src/values_extraction/leaves/mapping_var.rs

```diff
@@ -0,0 +1,258 @@
+    pub(crate) node: VectorWire<Target, { PAD_LEN(NODE_LEN) }>,
+    /// Leaf value
+    pub(crate) value: Array<Target, MAPPING_LEAF_VALUE_LEN>,
+    /// MPT root
+    pub(crate) root: KeccakWires<{ PAD_LEN(NODE_LEN) }>,
+    /// Storage mapping variable slot
+    pub(crate) slot: MappingSlotWires,
+    /// Identifier of the column of the table storing the key of the current mapping entry
+    pub(crate) key_id: Target,
+    /// Index denoting which EVM word are we looking at for the given variable
+    pub(crate) evm_word: Target,
+    #[serde(
+        serialize_with = "serialize_array",
+        deserialize_with = "deserialize_array"
+    )]
+    /// Boolean flags specifying whether the i-th column is a column of the table or not
+    pub(crate) is_actual_columns: [BoolTarget; MAX_COLUMNS],
+    #[serde(
+        serialize_with = "serialize_array",
+        deserialize_with = "deserialize_array"
+    )]
+    /// Boolean flags specifying whether the i-th field being processed has to be extracted into a column or not
+    pub(crate) is_extracted_columns: [BoolTarget; MAX_COLUMNS],
+    #[serde(
+        serialize_with = "serialize_long_array",
+        deserialize_with = "deserialize_long_array"
+    )]
+    /// Information about all columns of the table
+    pub(crate) table_info: [ColumnInfoTarget; MAX_COLUMNS],
+}
+
+/// Circuit to prove the correct derivation of the MPT key from a mapping slot
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct LeafMappingVarCircuit<
+    const NODE_LEN: usize,
+    const MAX_COLUMNS: usize,
+    const MAX_FIELD_PER_EVM: usize,
+> where
+    [(); PAD_LEN(NODE_LEN)]:,
+{
+    pub(crate) node: Vec<u8>,
+    pub(crate) slot: MappingSlot,
+    pub(crate) key_id: F,
+    pub(crate) evm_word: F,
+    pub(crate) num_actual_columns: usize,
+    pub(crate) num_extracted_columns: usize,
+    #[serde(
+        serialize_with = "serialize_long_array",
+        deserialize_with = "deserialize_long_array"
+    )]
+    pub(crate) table_info: [ColumnInfo; MAX_COLUMNS],
+}
+
+impl<const NODE_LEN: usize, const MAX_COLUMNS: usize, const MAX_FIELD_PER_EVM: usize>
+    LeafMappingVarCircuit<NODE_LEN, MAX_COLUMNS, MAX_FIELD_PER_EVM>
+where
+    [(); PAD_LEN(NODE_LEN)]:,
+{
+    pub fn build(
+        b: &mut CBuilder,
+    ) -> LeafMappingVarWires<NODE_LEN, MAX_COLUMNS, MAX_FIELD_PER_EVM> {
+        let zero = b.zero();
+
+        let key_id = b.add_virtual_target();
+        let evm_word = b.add_virtual_target();
+        let table_info = array::from_fn(|_| b.add_virtual_column_info());
+        let [is_actual_columns, is_extracted_columns] =
+            array::from_fn(|_| array::from_fn(|_| b.add_virtual_bool_target_safe()));
+
+        let slot = MappingSlot::mpt_key_with_offset(b, evm_word);
+
+        // Range check for the slot to restrict it's an Uint8.
+        b.range_check(slot.mapping_slot, 8);
+        // Range check for the EVM word to restrict it's an Uint32.
+        b.range_check(evm_word, 32);
```

Comment:
Same as for single variables circuit, we might skip these range checks since these values are hashed in the metadata digest